server/routes/one-time-code-auth/VerifyOneTimeCodeController.test.ts

```diff
@@ -88,28 +98,92 @@ describe('VerifyOneTimeCodeController', () => {
     it('should redirect to email sent page given no code submitted on request', async () => {
       req.body = {}
 
-      await verifyOneTimeCodeController.verifyOneTimeCode(req as undefined as Request, res as undefined as Response)
+      await verifyOneTimeCodeController.verifyOneTimeCode(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction
+      )
 
       expect(req.flash).toHaveBeenCalledWith('errors', [{ href: '#code', text: 'Enter the code from your email.' }])
       expect(res.redirect('email-sent'))
       expect(oneTimeCodeService.verifyOneTimeCode).not.toHaveBeenCalled()
       expect(req.session.barcodeUser).toEqual({ tokenValid: false, token: undefined })
     })
 
-    it('should redirect to request code page given API returns error indicating it could not verify the code', async () => {
+    it('should redirect to email sent page given API returns error indicating it could not verify the code', async () => {
       req.body = { code: 'ABCD' }
       oneTimeCodeService.verifyOneTimeCode.mockRejectedValue({
-        status: 400,
-        data: { status: 400, errorCode: { code: 'NOT_FOUND', userMessage: 'Not found' } },
+        status: 401,
+        data: { status: 401, errorCode: { code: 'OTC_NOT_FOUND', userMessage: 'Not found' } },
       })
 
-      await verifyOneTimeCodeController.verifyOneTimeCode(req as undefined as Request, res as undefined as Response)
+      await verifyOneTimeCodeController.verifyOneTimeCode(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction
+      )
+
+      expect(req.flash).toHaveBeenCalledWith('errors', [
+        { href: '#code', text: 'Enter the code we emailed you. This is 4 letters, like DNLC' },
+      ])
+      expect(res.redirect('email-sent'))
+      expect(req.session.barcodeUser).toEqual({ tokenValid: false, token: undefined })
+    })
+
+    it('should redirect to request code page given API returns error indicating there is no code to try for this session', async () => {
```

Comment:
Ah ha, the penny has just dropped re: the difference between OTC_NOT_FOUND and OTC_SESSION_NOT_FOUND - its about what we need the UI to do in respect of these errors. Got it! 😄